lib/galaxy_test/selenium/test_admin_app.py

```diff
@@ -1,3 +1,5 @@
+from selenium.webdriver.common.by import By
```

Comment:
This is not the path I would like to go with this at all. I've tried really hard to restrict Selenium imports - none of these tests depended on Selenium imports before. We should just add these now missing abstractions on to has_driver. That solution would be much more inline with the direction this code takes.

Reply:
I mainly wanted to check how difficult it'd be to move to Selenium 4.3.0. Feel free to take this over, since it also overlaps with https://github.com/galaxyproject/galaxy/pull/14188 .

Reply:
I've started work on top of #14188 @ https://github.com/jmchilton/galaxy/pull/new/dev_auto_update_dependencies.